src/pybind/mgr/dashboard/.pylintrc

```diff
@@ -119,7 +119,11 @@ disable=import-star-module-level,
         too-many-arguments,
         too-many-locals,
         too-many-statements,
-        useless-object-inheritance
+        useless-object-inheritance,
+        relative-beyond-top-level,
+        raise-missing-from,
+        super-with-arguments,
+        import-outside-toplevel
```

Comment:
Are these new `pylint` rules strictly enforcing what `isort` has fixed?

I'm asking this because I wouldn't add any fixer that we are not, on the other hand, enforcing via linting...

Otherwise, it's unlikely that contributors will spontaneously run the fixer... And we'll end up, from time to time, with code-shuffling PRs like this one, that add no core functionality but might potentially complicate backportings in ~100 files... I don't want to sound harsh, I really like automation like this one, but IMHO we should try to minimize as much as possible code shuffling, and if needed (as a side effect of increasing linting strictness) we should firmly ensure that the new rule is enforced on a PR basis.

Reply:
These lint errors appear with using `pylint 2.6.0`. So i added them to the `disable` list because i do not want to fix things that where OK for us with the old pylint version. These rules do not have anything in relation with introducing `isort`.

Reply:
> Otherwise, it's unlikely that contributors will spontaneously run the fixer... And we'll end up, from time to time, with code-shuffling PRs like this one, that add no core functionality but might potentially complicate backportings in ~100 files...

I fully agree that such a big change will complicate backporting. But at the moment our Python code is erroneous related to import order style. If we want to fix it, we need to go through this pain, it's not possible to make it by a smooth transition. If the rules are enabled, a lint process will throw the errors which will result in aborted `make check` runs. So all issues must be fixed nevertheless.

By the way, this PR was done because we have issues in the `[testenv:fix]` which makes me wondering that nobody has run this since ages.

> but IMHO we should try to minimize as much as possible code shuffling, and if needed (as a side effect of increasing linting strictness) we should firmly ensure that the new rule is enforced on a PR basis.

We can split this PR into separate ones. One that will increase `pylint` to the latest version. The latest `pylint` is a prereq of `isort`, because it introduces files that are used by `isort`.

I created 4 commits in this PR to separate each step, so it shouldn't be a problem to create separate PRs based on them.

Another PR will then introduce `isort` as another check in `[testenv:lint]` and `[testenv:fix]`.

Reply:
For now, and since this PR is not specific to this, I think we can let the lint and fix having different configurations.
We should create a issue and work on it soon.

Regarding the commits, I suggest changing it to either (I put them in order of preference):
1. Squash them all
This way we never get lint errors and commit title is correct
2. Squash them in 2 commits, one for `isort` and another for `pylint`, and change the title of the commits to mention only the one its changing.
This prevents having some commits that if checked out will generate lint errors.
3. Keep them all separated, but change the title of each commit and mention in some of them that linter will fail because fixes were implemented in another commit.

Reply:
Ok, so my question is: after this PR gets merge, if I add a new import in the wrong order, will I get my PR blocked?

Reply:
@epuertat `isort `is now called in the `lint` phase, so `make-check` will fail.
That being said, there are still some fixes that `autopep8 `does, that are not checked during `lint`

Reply:
Thanks! On the code-shuffling topic, [Black team](https://github.com/psf/black#migrating-your-code-style-without-ruining-git-blame) recommends using `.git-blame-ignore-revs`.

Reply:
> Ok, so my question is: after this PR gets merge, if I add a new import in the wrong order, will I get my PR blocked?

Yes it will, but we already have that behavior in the frontend where PRs are blocked when the Typescript is not formatted properly.

Running our scripts BEFORE committing a PR should be a standard behavior.

Reply:
>     2. Squash them in 2 commits, one for `isort` and another for `pylint`, and change the title of the commits to mention only the one its changing.
>        This prevents having some commits that if checked out will generate lint errors.

I think we should go this way.

Reply:
I know it's not directly related to the discussion... but would it be possible/make sense to check the stuff in /qa/tasks/mgr/dashboard as well?